### PR TITLE
Add CreateInvitedInstitutionController

### PR DESCRIPTION
**Pull request by brobsc**
**Feature/Bug description:** A new controller is necessary for the new mobile design on creating invited institutions.

**Solution:** Add the new controller, based on previous (desktop) controller.

criacao inst

**TODO/FIXME:** n/a

### CODE REVIEW

File: frontend/test/specs/institution/createInvitedInstitutionSpec.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    state.params.institutionKey = institution.key;
+
+    ctrl = $controller('CreateInvitedInstitutionController', {
+      scope,
+      authService,
+      institutionService,
+      imageService
+    });
+
+    // httpBackend.flush();
```
Comment (JuliePessoa) [inline frontend/test/specs/institution/createInvitedInstitutionSpec.js]: commented?

File: frontend/institution/createInvitedInstitutionController.js [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+           * @param {Image} src - current institutions image file (resized)
+           */
+          function saveProfileImage(src) {
+            if (!src) {
+              return Promise.resolve()
+            }
+
+            return ImageService.saveImage(src).then(data => {
+              ctrl.newInstitution.photo_url = data.url;
+              return Promise.resolve();
```
Comment (Luiz-FS) [inline frontend/institution/createInvitedInstitutionController.js]: Unnecessary promise

File: frontend/test/specs/institution/createInvitedInstitutionSpec.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  }));
+
+  describe('$onInit', () => {
+    it('should load the institution default data', () => {
+      spyOn(institutionService, 'getInstitution').and.callFake(() => {
```
Comment (JuliePessoa) [inline frontend/test/specs/institution/createInvitedInstitutionSpec.js]: couldn't just do` .callFake(() => $q.when(emptyInstitution)) `?

File: frontend/test/specs/institution/createInvitedInstitutionSpec.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        return {
+          then: callback => {
+            return callback(emptyInstitution);
+          },
+        }
+      })
+
+      ctrl.$onInit();
+      expect(ctrl.newInstitution).toEqual(emptyInstitution);
```
Comment (JuliePessoa) [inline frontend/test/specs/institution/createInvitedInstitutionSpec.js]: better mock `ctrl.loadInstitution` and then do a separated test for it

File: frontend/institution/createInvitedInstitutionController.js [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+           * as the User's current institution and update this info through the app and local storage.
+           * Then, sends the user back to STATE.HOME with a confirmation message.
+           */
+          function reloadAndRedirectHome() {
+            AuthService.reload().then(() => {
+              $state.go(STATES.HOME).then(() => {
+                ctrl.loading = false;
+                const alert = $mdDialog.alert({
+                  title: 'INSTITUIÇÃO CRIADA',
+                  textContent: 'Estamos processando suas permissões hierárquicas. Em breve você receberá uma notificação e ficará habilitado para administrar a instituição e toda sua hierarquia na Plataforma Virtual CIS.',
```
Comment (Luiz-FS) [inline frontend/institution/createInvitedInstitutionController.js]: Check type of invitation to display the message or not

**Review by JuliePessoa — CHANGES_REQUESTED** [review]

**Review by Luiz-FS — CHANGES_REQUESTED** [review]

**Review by JuliePessoa — APPROVED** [review]

**Review by Luiz-FS — APPROVED** [review]

**Pull request merged**